docs/applications_general.py

```diff
@@ -9,45 +9,39 @@
 #supporting modules (`gromacs_wrapper.py` and `lammps_wrapper.py`) 
 #wraps command line MD commands with a SCALE-MS wrapper layer.  The results should be in the same format.
 
-from gromacs_wrapper import collect_configurations, structure_to_pdb, make_input, simulate, modify_input
+from gromacs_wrapper import collect_configurations, internal_to_pdb, make_input, simulate, modify_input
 
 # configuration_list (an array of filenames) and md_inputs (a file
 # name, or list of file names if multiple inputs are needed to
 # intitialize the simulation.
 
-
-coordinate_inputs = configuration_input_list # some list of files
+# these lines are a BIT GROMACS specific, as other programs may have slightly different formats.
+coordinate_inputs = configuration_input_list # some list of files of length
 run_parameters = md_inputs # some list of files
-topology_files = topology_input_list # some list of files
+topology_files = topology_input_list # some list of files. 
+
+num_simulations = 1000
 
 initial_simulation_input = make_input(
     simulation_parameters=run_parameters,
     topology=topology_file,
     conformation=coordinate_inputs)
 
-# Set up an array of N simulations, starting from a single input.
-# MRS: seems like some instructions would need to be given here, such as:
-#      - which entries in the files to replaced by random numbers, or state dependent inputs
-#      - User should not have to worry about details such as whether temporary files would be needed should be 
-#      - so maybe the broadcasating would be done in make_input
-#      - 
-
-# a single modify commands but an array of modifiers.  If replacing some aspect of contents. 
-# ensemble obect.
-initial_input = scalems.broadcast(initial_simulation_input, shape=(N,))
+# Set up an array of num_simulations simulations, starting from a single input.
 
-# We will need a pdb for MSM building in PyEmma
-initial_pdb = coordinate_inputs[0]
+# NOTE: we presume that broadcast creates num_simulations even if the initial simulation input contains
+# fewer initial coordinates than num simulations. Like, if num_simulations is 40, and the initial 
+# coordinate list contains 7, then it will iterate (1,2,3,4,5,6,7,1,2,3,4,5, ... until reaching 40)?
+# and so on with the other files?
```

Comment:
That would not be consistent with typical meanings of "broadcast." In particular, I think we should have behavior as close to numpy as possible, though we may need to make a few decisions about implicit behavior or no-op sorts of things, but in general, I think that if you have an object of shape `(n,)` and you broadcast to dimensionality `(m,)` you would get an object of shape `(m,n)`, but we may want to make it an error unless the target dimensionality is either explicitly `(m,n)` or `(m,PLACEHOLDER)` where `PLACEHOLDER` is something we invent to stand in for a dimension size we don't know a priori. I think it is okay to let `(n,)` broadcast to `(m,n)` with a directive of an `m`-dimensional broadcast, but we can adopt a different syntax for the broadcast shape if that is confusing.

Reply:
OK, that makes sense I was just trying to understand the intended behavior in different situations.